src/XMakeBuildEngine/Evaluation/Evaluator.cs

```diff
@@ -2062,6 +2206,7 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
 
             foreach (string importFileEscaped in importFilesEscaped)
             {
+                atleastOneImportProcessed = true;
```

Comment:
Can you move this to line 2324, right after it gets added in the imports list? Then, it really means that at least one got processed. Otherwise rename boolean flag to atLeastOneFound.